### ISSUE TITLE
Currently unable to mock returns for functions with move only return values. e.g. unique_ptr

### ISSUE THREAD

**Issue by malcolmdavey** [eranpeer/FakeIt#281]
The following fails to compile

```
class A
{
public:
	virtual std::unique_ptr<std::string> method_move_only_return() = 0;
};

TEST(fakeit, Compile_Fails_ReturnMoveOnly)
{
	Mock<A> mockA;

	When(Method(mockA, method_move_only_return))
		.Return(make_unique<std::string>("value"));

	EXPECT_EQ("value", *mockA.get().method_move_only_return());
}
```

**Comment by malcolmdavey** [comment]
I have a solution for this one.

**Comment by FranckRJ** [comment]
Fixed by #282 in 2.3.1.
